### ISSUE TITLE
PostgreSQL BulkInsertOrUpdateAsync on entity with a default value causes InvalidOperationException

### ISSUE THREAD

**Issue by AeonLucid** [borisdj/EFCore.BulkExtensions#795]
Given the following code:

**Entity**
```csharp
public class ExampleEntity
{
    public Guid Id { get; set; }
    public int Field { get; set; }
    public bool Other { get; set; }
}
```

**Entity fluent configuration**
```csharp
builder.Property(x => x.UserId).ValueGeneratedNever();
builder.Property(x => x.Field).IsRequired();
builder.Property(x => x.Other).IsRequired().HasDefaultValue(false);
```

**Example**
```csharp
await context.BulkInsertOrUpdateAsync(new List<ExampleEntity>
{
    new ExampleEntity
    {
        Id = Guid.NewGuid(),
        Field = 1,
        Other = False
    },
    new ExampleEntity
    {
        Id = Guid.NewGuid(),
        Field = 1,
        Other = False
    }
});
```

The exception occurs:

```
System.InvalidOperationException: The binary import operation was started with X column(s), but X value(s) were provided.
```

This happens because fields with a default value in the database are excluded from the insert statement https://github.com/borisdj/EFCore.BulkExtensions/blob/abe6ca5ed73634bca19f47f8335a736d4ca5d119/EFCore.BulkExtensions/SQLAdapters/PostgreSql/PostgreSqlAdapter.cs#L68

**Comment by borisdj** [comment]
Currently that feature is supported only on SqlServer, will write a comment here once added for PG as well.

**Comment by AeonLucid** [comment]
For now I have just commented out the two lines which suits my usecase and makes InsertOrUpdate work. 

_Note, this is not related to the other issues regarding `PropertiesToExcludeOnUpdate`._

**Comment by xWiiLLz** [comment]
I'm also getting the same errors on version `6.4.2` (after PR #772 got introduced from [this issue](https://github.com/borisdj/EFCore.BulkExtensions/issues/769)). Downgrading to version `6.4.1` works fine though.

Perhaps the fix from #772 should be reverted since it's breaking this?

**Comment by rwasef1830** [comment]
PR #772 fix is incomplete, it doesn't handle case of Identity primary key. The implemented fix in that PR causes the property to not get skipped because it's a primary key which causes another exception, I will open a PR fixing that, as well fixing owned type support in postgresql (for case without table splitting).
